src/main/java/uk/gov/hmcts/reform/cpo/security/SecurityUtils.java

```diff
@@ -37,4 +50,38 @@ public String getServiceNameFromS2SToken(String serviceAuthenticationToken) {
     private String removeBearerFromToken(String token) {
         return token.startsWith(BEARER) ? token.substring(BEARER.length()) : token;
     }
+
+    public boolean hasCreatePermission() {
+        return hasPermission(CREATE);
+    }
+
+    public boolean hasReadPermission() {
+        return hasPermission(READ);
+    }
+
+    public boolean hasUpdatePermission() {
+        return hasPermission(UPDATE);
+    }
+
+    public boolean hasDeletePermission() {
+        return hasPermission(DELETE);
+    }
+
+    private boolean hasPermission(Permission permission) {
+        RequestAttributes requestAttributes = RequestContextHolder.getRequestAttributes();
+
+        String serviceAuthorizationHeaderValue = null;
+        if (requestAttributes != null) {
+            HttpServletRequest request = ((ServletRequestAttributes)requestAttributes).getRequest();
+            serviceAuthorizationHeaderValue = request.getHeader(SERVICE_AUTHORIZATION);
+        }
+
+        if (serviceAuthorizationHeaderValue == null) {
+            return false;
```

Comment:
This generally won't happen but can you add a log `warn` message before returning false.

Reply:
👍

Reply:
Fixed in https://github.com/hmcts/cpo-case-payment-orders-api/pull/68/commits/970b0c16e75dc6be385b97fa1c0dd42e08d20d2b